src/components/Checkbox/Checkbox.tsx

```diff
@@ -151,14 +169,22 @@ const CheckboxComponent: ForwardRefRenderFunction<HTMLInputElement, CheckboxProp
                                                   'tw-bg-box-disabled tw-text-box-disabled-inverse tw-border-line-strong',
                                           ])
                                         : merge([
+                                              !isCheckedOrMixed(state) && 'tw-bg-base hover:tw-bg-box-neutral',
```

Comment:
The cognitive complexity have increased here as well. It would be nice to simplify now that you added one more `emphasis` option.